### PR TITLE
fix(unplugin-vue-i18n): check for vue compiler in directive transform hook

### LINKED ISSUE #420: error when running Nuxt I18n module with latest Nuxt version (π version) in StackBlitz
### Reporting a bug?

Looks like Vite warmup calling `ssrTransform` directly without calling `config` hooks, `directive` Vite plugin here requires compiler from `vite:vue` plugin and it is missing or still not configured (maybe Nuxt bug).

Here a repo using `pnpm` to patch `@intlify/unplugin-vue-i18n/vite` (link to SB in the readme file):
https://github.com/userquin/nuxt-i18n-stackblitz-starter

### Expected behavior

The app should start but  sometimes doesn't work.

### Reproduction

https://stackblitz.com/fork/github/BobbieGoede/nuxt-i18n-starter/tree/v9

### Issue Package

unplugin-vue-i18n

### System Info

```shell
NA: works on local, the problem is StackBlitz

Anyway check the reproduction and the fix.
```

### Screenshot

imagen

Here the log patching `vite` manually:

imagen

### Additional context

I'm preparing a PR to fix this.

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/intlify/bundle-tools/blob/main/.github/contributing.md).
- [X] Read the README
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/intlify/bundle-tools/discussions).

### PR DESCRIPTION
**Pull request by userquin**
resolves #420

### CODE REVIEW

File: packages/unplugin-vue-i18n/src/core/directive.ts [hunk 1]
```diff
@@ -83,11 +83,13 @@ export function directivePlugin({
...
-            vuePluginOptions,
-            translationIdentifiers
-          )
+          if (vuePluginOptions?.compiler) {
+            analyzeIdentifiers(
+              getDescriptor(filename, code, vuePluginOptions),
+              vuePluginOptions,
+              translationIdentifiers
+            )
+          }
```
Comment (userquin) [inline packages/unplugin-vue-i18n/src/core/directive.ts:92]: should we reset `vuePluginOptions` if missing compiler?

Comment (kazupon) [reply packages/unplugin-vue-i18n/src/core/directive.ts:92]: Ah, yeah!
vue-i18n directive does not work without Vue compiler for SSR.

if Vue compiler is not specify on `vitejs/plugin-vue`, `unplugin-vue-i18n` should output warning.
(I recognize `vitejs/plugin-vue` has Vue compiler as defetault)

Comment (userquin) [reply packages/unplugin-vue-i18n/src/core/directive.ts:92]: Oh that's true, it is about the v-t directive, I'm going to check if ssr is exposed.

If this is only a vite plugin why not just expose the transform hook inside vite?

**Comment by userquin** [comment]
closing this, it is a bug in @vitejs/plugin-vue, compiler not being initialized correctly: https://github.com/vitejs/vite-plugin-vue/pull/475

**Comment by userquin** [comment]
@kazupon @BobbieGoede looks like Vite Vue plugin PR will took some time, I guess we can use the protection in this PR to avoid the error in Nuxt calling Vite warmup.

**Comment by BobbieGoede** [comment]
I agree we probably need to skip transform if the compiler is unset, it seems like both devtools (`Overlay.vue`) and nuxt (`nuxt-root.vue`) trigger this in some sort of race condition. I'm not sure if it's something we should/can fix at their end.

**Review by kazupon — APPROVED** [review]
LGTM!

**Pull request merged**